### DISCUSSION TITLE
BeginMoveDrag - operation is done in dispatcher

### DISCUSSION THREAD

**Discussion by KrzysztofDusko** [AvaloniaUI/Avalonia#14891] [General]
Hello.
In this PR https://github.com/AvaloniaUI/Avalonia/pull/14190 BeginMoveDrag action was moved to dispatcher.Post instead direct execution. This changind beharior in way that after call BeginMoveDrag  cotrol is immediatyly returen to caller..
Sample real word issue https://github.com/wieslawsoltes/Dock/issues/319

**Comment by kekekeks** [comment]
IIRC it's made for consistency with non-windows platforms where we the underlying platform doesn't introduce a nested event loop and doesn't send us "drag complete" notifications.

**Reply by KrzysztofDusko** [reply]
OK. But this introduce behavior change this is accepted in 11.1 ?

**Reply by timunie** [reply]
Crazy idea, would an async overload help you which is a Task? You could await it and then continue.

**Reply by kekekeks** [reply]
It's a _bug fix,_ since platforms were behaving inconsistently.
Such changes are acceptable for x.y.0 releases

**Reply by kekekeks** [reply]
>would an async overload help you which is a Task? 

We can't have one for non-windows platforms, unfortunately.

**Reply by maxkatz6** [reply]
> OK. But this introduce behavior change this is accepted in 11.1 ?

It's not a behavior change in general, since this behavior always existed on other platforms. We still might want to introduce another API that works better. `BeginResizeDrag/BeginMoveDrag` copied from WPF [are just broken](https://github.com/AvaloniaUI/Avalonia/issues/8429) (even on Windows, when touch comes to action).

In general, I always was of opinion of adding some attached property like `Window.DecorationType="Titlebar"` API (with other options like ResizeGrip and Minimize button), where platform would do visuals hit test (which we already do), and depending on attached property value it would start relevant operation associating with the pointer.

**Comment by KrzysztofDusko** [comment]
thanks for the answers.
